core/src/main/java/org/apache/calcite/adapter/enumerable/EnumerableValues.java

```diff
@@ -46,7 +46,7 @@
  * {@link org.apache.calcite.adapter.enumerable.EnumerableConvention enumerable calling convention}. */
 public class EnumerableValues extends Values implements EnumerableRel {
   /** Creates an EnumerableValues. */
-  private EnumerableValues(RelOptCluster cluster, RelDataType rowType,
+  EnumerableValues(RelOptCluster cluster, RelDataType rowType,
       ImmutableList<ImmutableList<RexLiteral>> tuples, RelTraitSet traitSet) {
```

Comment:
Always use `create` instead of new directly.

Reply:
Agree, calling directly the constructor is generally an anti-pattern. The problem here is that, in order to solve this bug, we need to propagate the traitSet from the `LogicalValues` to the `EnumerableValues`, at right now this is not possible via `create` method. I could define a new overloaded `create` method (I had done it previously in this PR and then removed it).

Reply:
Changed back: constructor is private again.